numba/tests/test_lists.py

```diff
@@ -1758,21 +1758,6 @@ def foo(x):
         for x in (-100, 100):
             self.assertEqual(foo.py_func(x), foo(x))
 
-    def test_not_unify(self):
-
-        @njit
-        def foo(x):
-            if x + 1 > 3:
-                l = ['a', 1, 2j]
-            else:
-                l = ['b', 2]
-            return l[0]
-
-        with self.assertRaises(errors.TypingError) as raises:
-            foo(100)
-        expect = "Cannot unify LiteralList"
-        self.assertIn(expect, str(raises.exception))
-
```

Comment:
This test should still work. I think this patch has had the side effect of specialising on the literal value of inputs:
```python
from numba import njit
import numpy as np

@njit
def foo(x):
    if x + 1 > 3:
        l = ['a', 1, 2j]
    else:
        l = ['b', 2]
    return l[0]

print(foo(100))  # 'a'
print(foo(-100)) # also 'a' !
```
note that `foo` will have the signature `(int64,)` but the IR will have over-specialised on the value `100`.

Reply:
The problem with this test case is that `typemap` holds information that `l[0] == 'a'`. I've put a check that ignores `static_getitem` but not sure if this is the best approach